pyOCD/tools/pyocd.py

```diff
@@ -218,8 +218,13 @@ def run(self):
                 else:
                     dumpHexData(data, args.read, width=args.width)
             elif args.action == ACTION_WRITE:
+                is_flash = self.isFlashWrite(args)
                 if args.width == 8:
-                    target.writeBlockMemoryUnaligned8(args.write, args.data)
+                    if is_flash:
+                        target.flash.init()
+                        target.flash.programPhrase(args.write, args.data)
```

Comment:
I was thinking as a future improvement you could simulate single byte writes by copying the current page, erasing it, and writing it back.  This will be needed for flash breakpoints.